### ISSUE TITLE
Support list_with_offset for Azure using continuation token

### ISSUE THREAD

**Issue by alexwilcoxson-rel** [apache/arrow-rs#5653]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
Azure object store implementation uses default `list_with_offset` implementation. It lists everything and does the filtering client side.

This is problematic for Delta Lake with large directories where we only really care about listing the latest files from a certain path.

**Describe the solution you'd like**
Microsoft pointed me to the [Hadoop Azure filesystem](https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java#L1289) `listStatus` API where `startFrom` is supported.

Basically the [starting path is embedded in the opaque continuation token](https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java#L1289) provided when the list API needs to page results. On the initial request this token is created and provided to the list API to start from the path.

The downside is that this is not documented. Perhaps this could be an opt-in option if there is hesitation using an undocumented format for the token that could break.

**Comment by tustvold** [comment]
I think I would prefer to not implement undocumented shenanigans, that I presume would be hard to test 😅

**Comment by alexwilcoxson-rel** [comment]
👍 we tried to implement this and what they suggested to us is not working anyway, so we're going to pursue other options.
